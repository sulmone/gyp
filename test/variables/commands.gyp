# This is a simple test file to make sure that variable substitution
# happens correctly.  Run "run_tests.py" using python to generate the
# output from this gyp file.

{
  'variables': {
    'pi': 'import math; print math.pi',
    'letters': 'ABCD',
  },
  'targets': [
    {
      'target_name': 'foo',
      'type': 'none',
      'variables': {
        'var1': '<!(["python", "-c", "<(pi)"])',
        'var2': '<!(python -c "print \'<!(python -c "<(pi)") <(letters)\'")',
        'var3': '<!(python -c "print \'<(letters)\'")',
        'var4': '<(<!(python -c "print \'letters\'"))',
      },
      'actions': [
        {
          'action_name': 'test_action',
          'inputs' : [
            '<(var2)',
          ],
          'outputs': [
            '<(var4)',
          ],
          'action': [
            'echo',
            '<(_inputs)',
            '<(_outputs)',
          ],
        },
      ],
    },
  ],
}
