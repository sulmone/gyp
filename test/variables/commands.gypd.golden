{'included_files': ['commands.gyp'],
 'targets': [{'actions': [{'action': ['echo',
                                      '"3.14159265359 ABCD"',
                                      'ABCD letters_list'],
                           'action_name': 'test_action',
                           'inputs': ['3.14159265359 ABCD'],
                           'outputs': ['ABCD', 'letters_list'],
                           'variables': {'var7': 'letters_list'}}],
              'configurations': {'Default': {}},
              'default_configuration': 'Default',
              'target_name': 'foo',
              'type': 'none',
              'variables': {'var1': '3.14159265359',
                            'var2': '3.14159265359 ABCD',
                            'var3': 'ABCD',
                            'var4': 'ABCD',
                            'var5': 'letters_',
                            'var6': 'list'}}],
 'variables': {'letters_list': 'ABCD', 'pi': 'import math; print math.pi'}}
